subprojects/core-api/src/main/java/org/gradle/internal/scripts/DefaultScriptFileResolver.java

```diff
@@ -48,13 +64,20 @@ public List<File> findScriptsIn(File dir) {
         }
         List<File> found = new ArrayList<File>(candidates.length);
         for (File candidate : candidates) {
-            if (candidate.isFile() && hasScriptExtension(candidate)) {
+            if (isCandidateFile(candidate) && hasScriptExtension(candidate)) {
```

Comment:
@bamboo Should we also notifying from this place?

Reply:
Not really. Eventually we will need to instrument the `dir.listFiles()` call above (somehow only considering script extensions) but that's for later.